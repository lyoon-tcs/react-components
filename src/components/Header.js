// dependencies
import React from 'react';
import PropTypes from 'prop-types';

const Header = (props) => {
  return <h1 className="Header">{props.title}</h1>;
};

Header.propTypes = {
  title: PropTypes.string
};

export default Header;
