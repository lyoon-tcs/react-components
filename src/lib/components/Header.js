/*
 * Copyright (c) 2006-present TrueChoice IP Holding Company, Inc.
 * All rights reserved.
 *
 * This software is the confidential and proprietary information of TrueChoice IP Holding Company, Inc.
 * ("Confidential Information").  You shall not disclose such Confidential Information and shall
 * use it only in accordance with the terms of the license agreement you entered into with the company.
 */

// dependencies
import React from 'react';
import PropTypes from 'prop-types';
// local files
import styles from '../../css/Header.module.css';

/**
 * @description - returns Header component
 * @param {string} props.title - the header's text
 * @param {string} props.heading - the heading (options: h1, h2, h3)
 * @return {jsx} - the Header component to render
 */
const Header = ({ title = 'Title', heading = 'h1' }) => {
  if (heading === 'h1') {
    return (
      <h1 style={{ fontSize: '40px' }} className={styles.header}>
        {title}
      </h1>
    );
  }
  if (heading === 'h2') {
    return (
      <h2 style={{ fontSize: '30px' }} className={styles.header}>
        {title}
      </h2>
    );
  }
  return (
    <h3 style={{ fontSize: '20px' }} className={styles.header}>
      {title}
    </h3>
  );
};

Header.propTypes = {
  title: PropTypes.string,
  heading: PropTypes.oneOf(['h1', 'h2', 'h3'])
};

export default Header;
