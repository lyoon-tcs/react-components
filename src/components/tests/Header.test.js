/*
 * Copyright (c) 2006-present TrueChoice IP Holding Company, Inc.
 * All rights reserved.
 *
 * This software is the confidential and proprietary information of TrueChoice IP Holding Company, Inc.
 * ("Confidential Information").  You shall not disclose such Confidential Information and shall
 * use it only in accordance with the terms of the license agreement you entered into with the company.
 */

// dependencies
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
// local files
import Header from '../Header';

afterEach(() => {
  cleanup();
});

describe('Header Component renders correctly', () => {
  const headerTitle = 'Hello world';

  test('should render Header with proper text', () => {
    render(<Header title={headerTitle} />);
    const headingElement = screen.getByText(headerTitle);
    expect(headingElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Header title={headerTitle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
