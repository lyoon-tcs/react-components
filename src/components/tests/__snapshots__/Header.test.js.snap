// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly matches snapshot 1`] = `
<h1
  className="Header"
>
  Hello world
</h1>
`;
